src/ir/enum_ty.rs

```diff
@@ -71,16 +71,20 @@ impl Enum {
         };
 
         declaration.visit(|cursor| {
-            if cursor.kind() == CXCursor_EnumConstantDecl {
```

Comment:
I sort of agree, and in general it uses to be good advice. But in this case, IMO at least, this is inverting the dependency instead.

That is, if for some other reason `enum_val_signed` returns `Some` when it's not a `CXCursor_EnumConstantDecl`, the code meaning is completely different, and we're in trouble.

We're traversing the ast looking for `EnumConstantDecl` nodes, so I think that `if` should stay there instead of relying that this only returns `Some` if that's the case.